### ISSUE TITLE
Can't connect to Win7 share since 1.41

### ISSUE THREAD

**Issue by cottonwool** [yoichiro/chromeos-filesystem-cifs#59]
Yesterday 16 June I could connect to a share on my Windows 7 Ultimate PC.
It worked well with a choice of shares offered when the connection was made. 
However, today I get 
net::ERR_CONNECTION_TIMED_OUT
every time.

I am using the host name and port 445 with no domain name needed.

What can I do to generate some useful debug from my chromebook?
I have limited skills but I can run Wireshark on my PC and I should be able to capture packets on my router.

**Comment by yoichiro** [comment]
@cottonwool Hmm... Could you give me the debug log on the console tab of the DevTools by the steps: https://github.com/yoichiro/chromeos-filesystem-cifs/wiki/How-to-Capture-the-Console-Log-of-DevTools

**Comment by cottonwool** [comment]
Console tab having tried a few times with machine name "rombald" and IP address. I've obfuscated the password as ****

background.js:6160 Object {type: "getSharedResources", serverName: "rombald", serverPort: "445", username: "mark", password: "***_"…}domainName: ""password: "**__"serverName: "rombald"serverPort: "445"type: "getSharedResources"username: "mark"**proto**: Object
background.js:6160 Client#login
background.js:6174 Domain name not specified. '?' will be applied.
background.js:6160 connect
background.js:6160 Object {type: "getSharedResources", serverName: "192.168.178.20", serverPort: "445", username: "mark", password: "_**_"…}
background.js:6160 Client#login
background.js:6174 Domain name not specified. '?' will be applied.
background.js:6160 connect
background.js:6160 Object {type: "getSharedResources", serverName: "192.168.178.20", serverPort: "445", username: "mark", password: "**__"…}
background.js:6160 Client#login
background.js:6174 Domain name not specified. '?' will be applied.
background.js:6160 connect
background.js:6160 Object {type: "getSharedResources", serverName: "rombald", serverPort: "445", username: "mark", password: "_***"…}
background.js:6160 Client#login
background.js:6174 Domain name not specified. '?' will be applied.
background.js:6160 connect

Watching with wireshark on the PC all I can see coming from the MAC address of my chromebook are ARP packets asking 
Who has 192.168.178.1? Tell 192.168.178.22
Who has 192.168.178.20? Tell 192.168.178.22
where .1 is the router, .20 the PC and .22 the chromebook.

Thanks very much.

**Comment by cottonwool** [comment]
Now using 1.43
A share is mounted, I'm offered a choice of shares (as expected) for my PC. However no files appear in the Files app for the share.

Looking at port 445 with Wireshark I can see SMB transactions 
381 1484.227070000  192.168.178.22  192.168.178.20  SMB 156 Trans2 Request, FIND_FIRST2, Pattern: \*

384 1484.231211000  192.168.178.20  192.168.178.22  SMB 1530    Trans2 Response, FIND_FIRST2, Files: . .. AlbumArtSmall.jpg alex-pro-summer.mp3 An Utterly Impartial History of Britain.aac BBC_R3_Drama_on_3_Piper_Alpha_06072008.mp3 books1 broadcastinghouse_20060423-1100_40_st.mp3 broadcastinghouse_20060430-1100_40_st.mp3 Catalogue.odt Covers desktop.ini digitalplanet_20060424-1700_40_st.mp3 Dr Who Folder.jpg gbtvwindows-61022-05-15-2007.wmv geekbrief-59453-05-03-2007.mp4 hd-59459-05-03-2007.mp4 iTunes moremusic.txt music1.txt music2.txt music3.txt music4.txt music5.txt My Albums My Playlists newsci-20060901-another-one-bites-the-moon-dust.mp3 Nokia Music Manager No_Pagliaccio_non_son.ogg Other Albums[Unreassembled Packet]

388 1484.234088000  192.168.178.20  192.168.178.22  SMB 2178    Trans2 Response<unknown>[transact continuation]

Then nothing apart from some TCP Keep Alive traffic

Looking at the background page console

Error in event handler for sockets.tcp.onReceive: RangeError: Offset is outside the bounds of the DataView
    at Types.getFixed4BytesValue (chrome-extension://mfhnnfciefdpolbelmfkpmhhmlkehbdf/background.js:6583:27)
    at ResponseUtils.loadData (chrome-extension://mfhnnfciefdpolbelmfkpmhhmlkehbdf/background.js:8104:47)
    at loadData (chrome-extension://mfhnnfciefdpolbelmfkpmhhmlkehbdf/background.js:9969:43)
    at FindFirst2Response.load (chrome-extension://mfhnnfciefdpolbelmfkpmhhmlkehbdf/background.js:9928:18)
    at Protocol.parseFindFirst2ResponsePacket (chrome-extension://mfhnnfciefdpolbelmfkpmhhmlkehbdf/background.js:11153:28)
    at null.<anonymous> (chrome-extension://mfhnnfciefdpolbelmfkpmhhmlkehbdf/background.js:12618:44)
    at null.<anonymous> (chrome-extension://mfhnnfciefdpolbelmfkpmhhmlkehbdf/background.js:11482:21)
    at ChromeSocket2.fetch (chrome-extension://mfhnnfciefdpolbelmfkpmhhmlkehbdf/background.js:11710:26)
    at _appendReceivedData (chrome-extension://mfhnnfciefdpolbelmfkpmhhmlkehbdf/background.js:11595:14)
handler @ extensions::uncaught_exception_handler:8
exports.handle @ extensions::uncaught_exception_handler:100
EventImpl.dispatch_ @ extensions::event_bindings:384
dispatchArgs @ extensions::event_bindings:247
dispatchEvent @ extensions::event_bindings:256

**Comment by yoichiro** [comment]
@cottonwool Could you try a new version 1.5.0?

**Comment by cottonwool** [comment]
Only had time to do a quick test with version 1.5
I started the tool, selected the remembered configuration and typed in the password.
The connection succeeded straight away and I was offered the choice of shares to connect to.
I then tried to copy several folders of pictures from my Chromebook to my PC. I was planning to do the copy anyway by USB memory stick, so this is real activity not a synthetic test.

489 files, 7 folders, 1.43GB (1,544,350,691 bytes) to be copied.

After copying most of the files the process seemed to hang up with the progress bar still not completed.

408 files, 7 folders 1.29 (1,385,943,938  bytes) were copied.

I didn't have any tracing happening or check the debug / console on the chromebook.

I will try this again when I have time and can report back some trace / debug and will report back.

**Comment by yoichiro** [comment]
@cottonwool At least, you could connect to your server with this app, right? If yes, I want to close this issue.

**Comment by cottonwool** [comment]
Yes I can connect to shares on my Windows 7 PC now 3 out of 4 times.
When I have connected I have been able to access files including viewing photos and listening to music.

**Comment by cottonwool** [comment]
I noticed in another thread that I can now specify the share by name instead of waiting to be given a choice to select from. So I've tried this out and I can open two shares at the same time by specifying the name of each share when I log-on. I like this feature. Also I can close one share and then open another one.
I can't reproduce the failure to mount I saw earlier today despite rebooting my Chromebook and  opening and closing shares and accessing files.

Good progress. Thanks. Mark.
